From c58cdd62e4e9372dab51300366d75d93d374f83b Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 13 Apr 2022 11:09:51 +0300
Subject: [PATCH 15/33] s32-cc: qspi: Use new compatible strings for s32g and
 s32r

8-8-8 DTR mode has different settings on S32G and S32R. For this
purpose a new set of compatible strings are needed.

Issue: ALB-8592
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/dts/fsl-s32-gen1.dtsi | 1 -
 arch/arm/dts/fsl-s32g.dtsi     | 1 +
 arch/arm/dts/fsl-s32r45.dts    | 1 +
 arch/arm/mach-s32/s32-cc/soc.c | 2 +-
 drivers/spi/fsl_qspi.c         | 3 ++-
 5 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/arch/arm/dts/fsl-s32-gen1.dtsi b/arch/arm/dts/fsl-s32-gen1.dtsi
index d51f490697..bc6b7d7344 100644
--- a/arch/arm/dts/fsl-s32-gen1.dtsi
+++ b/arch/arm/dts/fsl-s32-gen1.dtsi
@@ -294,7 +294,6 @@
 	};
 
 	qspi: quadspi@40134000 {
-		compatible = "fsl,s32cc-qspi";
 		#address-cells = <1>;
 		#size-cells = <0>;
 		reg = <0x0 0x00000000 0x0 0x20000000>,
diff --git a/arch/arm/dts/fsl-s32g.dtsi b/arch/arm/dts/fsl-s32g.dtsi
index 15eead108b..2ba3c9af87 100644
--- a/arch/arm/dts/fsl-s32g.dtsi
+++ b/arch/arm/dts/fsl-s32g.dtsi
@@ -205,6 +205,7 @@
 };
 
 &qspi {
+	compatible = "nxp,s32g-qspi", "nxp,s32cc-qspi";
 	status = "okay";
 
 	mx25uw51245g@0 {
diff --git a/arch/arm/dts/fsl-s32r45.dts b/arch/arm/dts/fsl-s32r45.dts
index 195cdb5466..b853df9fff 100644
--- a/arch/arm/dts/fsl-s32r45.dts
+++ b/arch/arm/dts/fsl-s32r45.dts
@@ -170,6 +170,7 @@
 };
 
 &qspi {
+	compatible = "nxp,s32r45-qspi", "nxp,s32cc-qspi";
 	pinctrl-0 = <&pinctrl0_qspi>;
 	pinctrl-names = "default";
 	status = "okay";
diff --git a/arch/arm/mach-s32/s32-cc/soc.c b/arch/arm/mach-s32/s32-cc/soc.c
index 6d9fc72ac6..b744b47fbc 100644
--- a/arch/arm/mach-s32/s32-cc/soc.c
+++ b/arch/arm/mach-s32/s32-cc/soc.c
@@ -80,7 +80,7 @@ static void disable_qspi_mmu_entry(void)
 	size_t i;
 
 	offset = fdt_node_offset_by_compatible(gd->fdt_blob, -1,
-					       "fsl,s32cc-qspi");
+					       "nxp,s32cc-qspi");
 	if (offset > 0) {
 		if (fdtdec_get_is_enabled(gd->fdt_blob, offset))
 			return;
diff --git a/drivers/spi/fsl_qspi.c b/drivers/spi/fsl_qspi.c
index a9f84d5f4b..aab1d5671b 100644
--- a/drivers/spi/fsl_qspi.c
+++ b/drivers/spi/fsl_qspi.c
@@ -1268,7 +1268,8 @@ static const struct udevice_id fsl_qspi_ids[] = {
 	{ .compatible = "fsl,imx6ul-qspi", .data = (ulong)&imx6ul_7d_data },
 	{ .compatible = "fsl,imx7d-qspi", .data = (ulong)&imx6ul_7d_data },
 	{ .compatible = "fsl,imx7ulp-qspi", .data = (ulong)&imx7ulp_data },
-	{ .compatible = "fsl,s32cc-qspi", .data = (ulong)&s32cc_data },
+	{ .compatible = "nxp,s32g-qspi", .data = (ulong)&s32cc_data },
+	{ .compatible = "nxp,s32r45-qspi", .data = (ulong)&s32cc_data },
 	{ }
 };
 
-- 
2.17.1

